static/app/components/codecov/container/codecovParamsProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import {useCallback, useEffect} from 'react';
+
+import type {CodecovContextData} from 'sentry/components/codecov/context/codecovContext';
+import {CodecovContext} from 'sentry/components/codecov/context/codecovContext';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type CodecovProviderProps = {
+  children?: NonNullable<React.ReactNode>;
+};
+
+export default function CodecovQueryParamsProvider({children}: CodecovProviderProps) {
+  const organization = useOrganization();
+  const orgSlug = organization.slug;
+
+  const location = useLocation();
+  const [localStorageState, setLocalStorageState] = useLocalStorageState(
+    `codecov-selection:${orgSlug}`,
+    {} as Partial<CodecovContextData>
+  );
+  const navigate = useNavigate();
+
+  const updateSelectorData: CodecovContextData['updateSelectorData'] = useCallback(
+    data => {
+      const currentParams = new URLSearchParams(location.search);
+      currentParams.set(Object.keys(data)[0] as string, Object.values(data)[0] as string);
+      navigate(`${location.pathname}?${currentParams.toString()}`, {replace: true});
+    },
+    [location.search, location.pathname, navigate]
+  );
+
+  useEffect(() => {
+    const repository =
+      typeof location.query.repository === 'string' ? location.query.repository : null;
+    const integratedOrg =
+      typeof location.query.integratedOrg === 'string'
+        ? location.query.integratedOrg
+        : null;
+    const branch =
+      typeof location.query.branch === 'string' ? location.query.branch : null;
+    const codecovPeriod =
+      typeof location.query.codecovPeriod === 'string'
+        ? location.query.codecovPeriod
+        : null;
+
+    setLocalStorageState({repository, integratedOrg, branch, codecovPeriod});
+  }, [setLocalStorageState, location.query]);
+
+  const params: CodecovContextData = {
+    repository:
+      decodeURIComponent(location.query.repository as string).trim() ||
```

Comment:
Why is .trim() required?

Reply:
In case the values I get have whitespace in a future. Technically I control the input so I know it isn't happening today, so this is more of a future proof case. 

Technically adding complexity now for robustness later, so I'm okay getting rid of it today and adding it later if we anticipate strings with whitespaces

Reply:
Why would it get a whitespace, and why would the whitespace be invalid?

Reply:
I suppose there wouldn't be for now since the value is controlled by data we manipulate, not user data. And even if so, it should be okay.